sdk/provisioning/Azure.Provisioning/src/ResourceOfT.cs

```diff
@@ -24,7 +24,20 @@ public abstract class Resource<T> : Resource
         /// <summary>
         /// Gets the properties of the resource.
         /// </summary>
-        public T Properties { get; }
+        public T Properties
+        {
+            get
+            {
+                if (IsExisting)
+                {
+                    throw new InvalidOperationException("Properties are not available for existing resources");
```

Comment:
lets add a future item to be able to do a GET on the resource so that the properties might be available.

Reply:
properties are actually available for outputs.